sdk/src/main/scala/com/horizen/account/state/ForgerStakeMsgProcessor.scala

```diff
@@ -365,105 +312,65 @@ case class ForgerStakeMsgProcessor(params: NetworkParams) extends AbstractFakeSm
+  def doRemoveStakeCmd(msg: Message, view: BaseAccountStateView): Array[Byte] = {
     val inputParams = getArgumentsFromData(msg.getData)
-    val cmdInput = RemoveStakeCmdInputDecoder.decode(inputParams) match {
-      case Success(obj) => obj
-      case Failure(exception) =>
-        val msgStr = "Error while parsing cmd input"
-        log.debug(msgStr, exception)
-        return new ExecutionFailed(AddNewStakeGasPaidValue, new IllegalArgumentException(exception))
-    }
-
+    val cmdInput = RemoveStakeCmdInputDecoder.decode(inputParams)
     val stakeId: Array[Byte] = cmdInput.stakeId
     val signature: SignatureSecp256k1 = cmdInput.signature
 
     // get the forger stake data to remove
-    val stakeData = findStakeData(view, stakeId) match {
-      case Some(obj) => obj
-      case None =>
-        val msgStr = "No such stake id in state-db"
-        log.debug(msgStr)
-        return new ExecutionFailed(RemoveStakeGasPaidValue, new Exception(msgStr))
-    }
+    val stakeData = findStakeData(view, stakeId)
+      .getOrElse(throw new ExecutionFailedException("No such stake id in state-db"))
 
     // check signature
     val msgToSign = getMessageToSign(stakeId, msg.getFrom.address(), msg.getNonce.toByteArray)
     if (!signature.isValid(stakeData.ownerPublicKey, msgToSign)) {
-      val msgStr = "Invalid signature"
-      log.debug(msgStr)
-      return new ExecutionFailed(RemoveStakeGasPaidValue, new Exception(msgStr))
+      throw new ExecutionFailedException("Invalid signature")
     }
 
     // remove the forger stake data
     removeForgerStake(view, stakeId)
 
     val removeStakeEvt = WithdrawForgerStake(stakeData.ownerPublicKey, stakeId)
     val evmLog = getEvmLog(removeStakeEvt)
-    view.addLog(evmLog).get
+    view.addLog(evmLog)
 
     // decrease the balance of the "stake smart contract” account
-    view.subBalance(fakeSmartContractAddress.address(), stakeData.stakedAmount).get
+    view.subBalance(fakeSmartContractAddress.address(), stakeData.stakedAmount)
 
     // increase the balance of owner (not the sender) by withdrawn amount.
     view.addBalance(stakeData.ownerPublicKey.address(), stakeData.stakedAmount)
 
     // Maybe result is not useful in case of success execution (used probably for RPC commands only)
-    val result = stakeId
-    new ExecutionSucceeded(RemoveStakeGasPaidValue, result)
+    stakeId
   }
 
-  override def process(msg: Message, view: BaseAccountStateView): ExecutionResult = {
-    try {
-
-      if (!canProcess(msg, view)) {
-        val errMsg = s"Cannot process message $msg"
-        log.error(errMsg)
-        return new InvalidMessage(new IllegalArgumentException(errMsg))
-      }
-
-      val cmdString = BytesUtils.toHexString(getOpCodeFromData(msg.getData))
-      cmdString match {
-        case `GetListOfForgersCmd` =>
-          doGetListOfForgersCmd(msg, view)
+  override def process(msg: Message, view: BaseAccountStateView, gas: GasPool): Array[Byte] = {
```

Comment:
With automatic gas consumption the same operation does not necessarily have a constant gas cost anymore, as the amount of operations depends on the current state. I enabled automatic gas consumption, but disabled gas checks in the tests because of that for now. I will open a separate PR to reenable that and fix the tests.